### DISCUSSION TITLE
ShowModal

### DISCUSSION THREAD

**Discussion by Roboteco** [Embarcadero/DelphiVCL4Python#53] [Q&A] [answered]
Hello!

I don't understand how to return the result of Modal Form

In Forn1:
    def __on_buttonCancel_click(self, sender):
        self.Result = mrCancel
        self.Close()

...
print(form1.ShowModal)   # return 2
...

**Comment by lmbelo** [comment, marked as answer]
The ShowModal method has a return value, it is the same as the ModalResult property:

```
modal_result = MyForm.ShowModal()   
if modal_result == mrOk:
    ...
```



https://embarcadero.github.io/DelphiVCL4Python/reference/delphivcl.CustomForm.html#delphivcl.CustomForm.ShowModal
